### PR TITLE
Feature/436 usdf hotfix

### PR DESCRIPTION
**Pull request by arnabmitra**
## Description

The 1.5.0 version of provenanced does not correctly support creation of restricted markers via governance proposal. A restricted marker is required for the USDF stable coin. Unfortunately due to scheduling concerns the 1.6.0 release of provenanced that correctly resolves these issues (see #431, #430) can not be used.
Additionally this uses the upgrade handler to update any existing `usdf.c` to the right `marker_type`

closes: #436

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by arnabmitra** [comment]
additional testing notes
original state:


provenance on  tags/v1.5.0 [$!] at ☸️ gke_figure-development_us-east1-b_tf-test took 5s
➜ provenanced -t query marker get usdf.c
marker:
  '@type': /provenance.marker.v1.MarkerAccount
  access_control: []
  allow_governance_control: true
  base_account:
    account_number: "11"
    address: tp1c9pv3n0pmjezg0wkgwqze00957ve39hwsjy84h
    pub_key: null
    sequence: "0"
  denom: usdf.c
  manager: tp125rltkc8kmkm4hcdnl7uda9deau9aldryh089c
  marker_type: MARKER_TYPE_COIN
  status: MARKER_STATUS_PROPOSED
  supply: "0"
  supply_fixed: false
after "usdf.c-hotfix"

➜ provenanced -t query marker get usdf.c
marker:
  '@type': /provenance.marker.v1.MarkerAccount
  access_control: []
  allow_governance_control: true
  base_account:
    account_number: "11"
    address: tp1c9pv3n0pmjezg0wkgwqze00957ve39hwsjy84h
    pub_key: null
    sequence: "0"
  denom: usdf.c
  manager: tp125rltkc8kmkm4hcdnl7uda9deau9aldryh089c
  marker_type: MARKER_TYPE_RESTRICTED
  status: MARKER_STATUS_PROPOSED
  supply: "0"
  supply_fixed: false
@iramiller we should probably gov propose usdc.fon pio-test so that it runs there before prod

File: x/marker/keeper/keeper_test.go [hunk 1]
```diff
@@ -128,6 +128,81 @@ func TestAccountKeeperReader(t *testing.T) {
...
+		[]types.AccessGrant{*types.NewAccessGrant(user, []types.Access{types.Access_Mint})})
+
+	require.NoError(t, app.MarkerKeeper.AddMarkerAccount(ctx, mac))
+
+	m, err := app.MarkerKeeper.GetMarkerByDenom(ctx, "usdf.c")
+	require.NoError(t, err)
+	require.NotNil(t, m)
+	require.EqualValues(t, m.GetDenom(), "usdf.c")
+	require.EqualValues(t, m.GetAddress(), addr)
+
```
Comment (iramiller) [inline x/marker/keeper/keeper_test.go:151]: We don't really need lines 151 through 156 as the get by denom and get by address checks are not needed here.

File: x/marker/keeper/keeper_test.go [hunk 2]
```diff
@@ -128,6 +128,81 @@ func TestAccountKeeperReader(t *testing.T) {
...
+	user := testUserAddress("test")
+	// create account and check default values
+	mac := types.NewEmptyMarkerAccount(
+		"testcoin",
+		user.String(),
+		[]types.AccessGrant{*types.NewAccessGrant(user, []types.Access{types.Access_Mint})})
+
+	require.NoError(t, app.MarkerKeeper.AddMarkerAccount(ctx, mac))
+
+	m, err := app.MarkerKeeper.GetMarkerByDenom(ctx, "testcoin")
```
Comment (iramiller) [inline x/marker/keeper/keeper_test.go:183]: similar to above I don't think we need to compare get by address/get by denom in this test or count up the instances of a marker with the given primary key.

**Review by iramiller — COMMENTED** [review]
We can streamline these tests a bit to reduce the size of this PR to the absolute minimum.

File: x/marker/keeper/keeper_test.go (hunk 1 above)
Comment (arnabmitra) [reply x/marker/keeper/keeper_test.go:151]: removed

File: x/marker/keeper/keeper_test.go (hunk 2 above)
Comment (arnabmitra) [reply x/marker/keeper/keeper_test.go]: removed

**Review by iramiller — APPROVED** [review]
LGTM

**Pull request merged**